src/android/com/adobe/phonegap/push/GCMIntentService.java

```diff
@@ -119,6 +122,49 @@ private void replaceKey(String oldKey, String newKey, Bundle extras, Bundle newE
     }
 
     /*
+     * Normalize localization for key
+     */
+    private String localizeKey(Context context, String key, String value) {
+        if (key.equals(TITLE) || key.equals(MESSAGE) || key.equals(SUMMARY_TEXT)) {
+            try {
+                JSONObject localeObject = new JSONObject(value);
```

Comment:
Maybe use this (similar method [.mayBeJSON(String str)](http://json-lib.sourceforge.net/apidocs/jdk15/net/sf/json/util/JSONUtils.html#mayBeJSON%28java.lang.String%29)  available in the JSONUtils library):

```java
if (value.startWith("{") && value.endWith("}")
    && value.toLowerCase().contains("locKey".toLowerCase()) {
  // do stuff
}
```

I doubt that users show the string containing this, but may be exceptions.